tmail-backend/backend-commons/elasticsearch-v6/src/main/java/org/apache/james/backends/es/ElasticSearchIndexer.java

```diff
@@ -83,6 +83,10 @@ private void logContent(DocumentId id, String content) {
     }
 
     public Mono<BulkResponse> update(List<UpdatedRepresentation> updatedDocumentParts, RoutingKey routingKey) {
+        return update(updatedDocumentParts, routingKey, false);
+    }
+
+    public Mono<BulkResponse> update(List<UpdatedRepresentation> updatedDocumentParts, RoutingKey routingKey, boolean upsert) {
```

Comment:
I remember that having a boolean in a method's arguments is a code smell that we should have 2 different methods instead (here upsert one and just update one).

Reply:
I do have two methods?

Reply:
```
public Mono<BulkResponse> update(List<UpdatedRepresentation> updatedDocumentParts, RoutingKey routingKey)

public Mono<BulkResponse> upsert(List<UpdatedRepresentation> updatedDocumentParts, RoutingKey routingKey)
```

Maybe one private to avoid duplicate code.
```
private Mono<BulkResponse> update(List<UpdatedRepresentation> updatedDocumentParts, RoutingKey routingKey, boolean upsert)
```

Reply:
Ah get it :) Good idea

Reply:
Done!